duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -45,12 +46,19 @@ public Http(String in_method, String in_host, String in_uri) {
         headers = new Headers.Builder();
         headers.add("Host", host);
 
+        httpClient = new OkHttpClient();
         proxy = null;
+        if (proxy != null) {
+            httpClient.setProxy(proxy);
+        }
+        httpClient.setConnectTimeout(timeout, TimeUnit.SECONDS);
+        httpClient.setWriteTimeout(timeout, TimeUnit.SECONDS);
+        httpClient.setReadTimeout(timeout, TimeUnit.SECONDS);
     }
 
     public Http(String in_method, String in_host, String in_uri, int timeout) {
-      this(in_method, in_host, in_uri);
-      this.timeout = timeout;
+        this(in_method, in_host, in_uri);
+        this.timeout = timeout;
```

Comment:
I don't think this will work if a timeout different from the default is provided - this will first call the constructor at line 41, which will set (lines 54-56) the default timeout onto the httpClient, then the timeout will be overwritten on line 61 - but it's now too late.

Reply:
You are correct.  Good catch.

I will update